utilities/bulk-string-replace.py

```diff
@@ -0,0 +1,179 @@
+    print "Fetching prefixes from NIPAP... ",
+    sys.stdout.flush()
+    n = 1
+    prefix_list = []
+    t0 = time.time()
+    query = {
+        'operator': 'or',
+        'val1': {
+            'operator': 'regex_match',
+            'val1': 'description',
+            'val2': pattern
+        },
+        'val2': {
+            'operator': 'regex_match',
+            'val1': 'node',
+            'val2': pattern
+        }
+    }
+    full_result = Prefix.search(query, { 'parents_depth': -1, 'max_result': BATCH_SIZE })
+    prefix_result = full_result['result']
+    prefix_list += prefix_result
+    print len(prefix_list), 
+    sys.stdout.flush()
+    while len(prefix_result) == 100:
+        full_result = Prefix.smart_search(pattern, { 'parents_depth': -1, 'max_result': BATCH_SIZE, 'offset': n * BATCH_SIZE })
+        prefix_result = full_result['result']
+        prefix_list += prefix_result
+        print len(prefix_list), 
+        sys.stdout.flush()
+        n += 1
+
+    t1 = time.time()
+    print " done in %.1f seconds" % (t1 - t0)
+
+    # Display list
+    print_pattern = "%-2s%-14s%-2s%-30s%-20s%s"
+    print "\n\nPrefixes to change:"
+    print print_pattern % ("", "VRF", "", "Prefix", "Node", "Description")
+    i_match = 0
+    for i, prefix in enumerate(prefix_list):
+        if prefix.match:
+            if use_colors:
+                print "\x1b[0m",
+            print " -- %d --" % i
+            color = "\x1b[91m"
+        else:
+            color = "\x1b[0m"
+            
+        print (color + print_pattern) % (
+            "-" if prefix.match else "",
+            prefix.vrf.rt,
+            prefix.type[0].upper(),
+            (("  " * prefix.indent) + prefix.display_prefix)[:min([ len(prefix.display_prefix) + 2*prefix.indent, 30 ])],
+            (prefix.node or '')[:min([ len(prefix.node or ''), 20 ])],
+            (prefix.description or '')[:min([ len(prefix.description or ''), 900 ])]
+        )
+        if prefix.match:
+            new_prefix_node = re.sub(pattern, replacement, (prefix.node or ''), flags=re.IGNORECASE)
+            new_prefix_desc = re.sub(pattern, replacement, (prefix.description or ''), flags=re.IGNORECASE)
+            print ("\x1b[92m" + print_pattern) % (
+                "+",
+                prefix.vrf.rt,
+                prefix.type[0].upper(),
+                ("  " * prefix.indent + prefix.display_prefix)[:min([ len(prefix.display_prefix) + 2*prefix.indent, 30 ])],
+                new_prefix_node[:min([ len(new_prefix_node), 20 ])],
+                new_prefix_desc[:min([ len(new_prefix_desc), 90 ])]
+            )
+
+
+    # reset colors
+    print "\x1b[0m",
+
+    # Perform action?
+    print "Select replacements to perform"
+    print "Enter comma-separated selection (eg. 5,7,10) or \"all\" for all prefixes."
+    print "Prefix list with ! to invert selection (eg !5,7,10 to perform operation on all except the entered prefixes)"
+    inp = raw_input("Selection: ").strip()
```

Comment:
Yeah, ^C was the intention (even tough it throws a nasty traceback in the user's face).

I wanted to avoid input as that's the name of a built-in function, see https://docs.python.org/2/library/functions.html#input. I could probably have chosen a better name though.